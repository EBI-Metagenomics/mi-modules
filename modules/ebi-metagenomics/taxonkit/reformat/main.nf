process TAXONKIT_REFORMAT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/taxonkit:0.17.0--h9ee0642_1':
        'biocontainers/taxonkit:0.17.0--h9ee0642_1' }"

    input:
    tuple val(meta), path(tsv)
    path taxdb

    output:
    tuple val(meta), path("*.tsv"), emit: reformat_tsv
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    taxonkit \\
        reformat \\
        $args \\
        --threads $task.cpus \\
        --data-dir $taxdb \\
        --out-file ${prefix}.tsv \\
        $tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        taxonkit: \$( taxonkit version | sed 's/.* v//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        taxonkit: \$( taxonkit version | sed 's/.* v//' )
    END_VERSIONS
    """
}
