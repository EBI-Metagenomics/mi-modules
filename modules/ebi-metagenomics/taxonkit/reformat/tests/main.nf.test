nextflow_process {

    name "Test Process TAXONKIT_REFORMAT"
    script "../main.nf"
    process "TAXONKIT_REFORMAT"

    tag "modules"
    tag "modules_nfcore"
    tag "taxonkit"
    tag "taxonkit/reformat"

    config "./nextflow.config"

    test("tsv with header") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("$moduleTestDir/fixtures/nucl_gb.accession2taxid", checkIfExists: true)
                ]
                input[1] = file("$moduleTestDir/fixtures/taxdump", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("tsv with header - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("$moduleTestDir/fixtures/nucl_gb.accession2taxid", checkIfExists: true)
                ]
                input[1] = file("$moduleTestDir/fixtures/taxdump", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
