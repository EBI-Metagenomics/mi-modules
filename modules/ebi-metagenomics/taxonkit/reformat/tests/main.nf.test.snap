{
    "tsv with header": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,331e1290ca8ae7e3962b43e28d865bae"
                    ]
                ],
                "1": [
                    "versions.yml:md5,25e183506a3cbe3ddc98e6036b4a0525"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,331e1290ca8ae7e3962b43e28d865bae"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,25e183506a3cbe3ddc98e6036b4a0525"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-15T10:21:38.67633"
    },
    "tsv with header - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,25e183506a3cbe3ddc98e6036b4a0525"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,25e183506a3cbe3ddc98e6036b4a0525"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-15T10:21:47.486344"
    }
}